File: 251031/Rixsiyev/project/project 3/src/main/java/com/example/entitiesapp/model/Mark.java
package com.example.entitiesapp.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "tbl_mark")
@Data
@EqualsAndHashCode(callSuper = false)
public class Mark implements BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @ManyToMany(mappedBy = "marks")
    private List<Topic> topics = new ArrayList<>();
} 

File: 251003/Kukhatskavolets/microblog/publisher/src/main/kotlin/by/kukhatskavolets/publisher/entities/Mark.kt
package by.kukhatskavolets.publisher.entities

import jakarta.persistence.*

@Entity
@Table(name = "tbl_mark", schema = "public")
data class Mark(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,

    @Column(nullable = false, unique = true, length = 32)
    val name: String  = "",
)